com.unity.shadergraph/Editor/Data/Nodes/Utility/CustomFunctionNode.cs

```diff
@@ -385,26 +393,33 @@ void ValidateBareTextureSlots()
 
         public override void ValidateNode()
         {
+            bool hasAnyOutputs = this.GetOutputSlots<MaterialSlot>().Any();
             if (sourceType == HlslSourceType.File)
             {
+                SourceFileStatus fileStatus = SourceFileStatus.Empty; 
                 if (!string.IsNullOrEmpty(functionSource))
                 {
                     string path = AssetDatabase.GUIDToAssetPath(functionSource);
                     if (!string.IsNullOrEmpty(path))
                     {
                         string extension = path.Substring(path.LastIndexOf('.'));
                         if (!s_ValidExtensions.Contains(extension))
-                        {
-                            owner.AddValidationError(objectId, k_InvalidFileType, ShaderCompilerMessageSeverity.Error);
-                        }
+                            fileStatus = SourceFileStatus.Invalid;
                         else
-                        {
-                            owner.ClearErrorsForNode(this);
-                        }
+                            fileStatus = SourceFileStatus.Valid;
                     }
```

Comment:
Any reason to remove the curly braces here?  (I'm not asking to put them back, but have a mild preference for them)

Reply:
No reason, it's just how it ended up after refactoring the logic a few times